### ISSUE TITLE
Clicking on notifications does not mark them as read.

### ISSUE THREAD

**Issue by billw2012** [Nexus-Mods/web-issues#1325]
## Bug/error being reported:
Clicking on notifications doesn't mark them as read, in the new system OR the old one. However *middle clicking* them does.

## Steps to reproduce:
Click a notification.

## Expected result:
It marks the notification as read.

## Actual result:
It doesn't mark the notification as read.

**Comment by Pickysaurus** [comment]
Do you get any kind of error when clicking on them?

**Comment by Pickysaurus** [comment]
Please also provide the name of your browser and any addons you are using. I cannot replicate this.

**Comment by billw2012** [comment]
Sorry I intended to both add browser and a gif showing it, but got distracted!

Browser is Firefox 85.0.2 (64-bit).
Me clicking notification:
https://i.imgur.com/UoAlTJl.gif
It goes to the notification, despite the error, but doesn't clear it.
Me middle clicking notification:
https://i.imgur.com/lUdRXNC.gif
It correctly opens another tab to the item, and marks it read.

I have a few plugins installed (ublock, dark reader and stylus are disabled for your pages):
![image](https://user-images.githubusercontent.com/1453936/112143934-722de180-8bd0-11eb-8f52-5233bc703693.png)

**Comment by billw2012** [comment]
Also, if you post another message for me here https://www.nexusmods.com/valheim/mods/616/?tab=posts I will click it in notifications and then copy the browser log for you.

**Comment by Pickysaurus** [comment]
Thanks for the info. We've confirmed this is a bug in Firefox itself, as documented here: https://bugzilla.mozilla.org/show_bug.cgi?id=1280189

In basic terms, Firefox cuts off any pending network requests on page reload, causing the request to mark the notification as read to fail. We're looking at a possible workaround, but this is sadly a flaw on Mozilla's end.

**Comment by billw2012** [comment]
Surely you just make the click mark the notification as read, wait for that to complete, then navigate?
Or look at what github itself is doing. I just tested and it does the correct behaviour from the notifications page: you click the notification and it both navigates to the page, and marks it read, and works in my browser.

**Comment by Pickysaurus** [comment]
> Surely you just make the click mark the notification as read, wait for that to complete, then navigate?
That is one option. Be aware though this is an issue **only** with Firefox. I often compare it to IE in terms of general compatibility with any web technologies. 
> Or look at what github itself is doing. I just tested and it does the correct behaviour from the notifications page: you click the notification and it both navigates to the page, and marks it read, and works in my browser.
There is no longer a notifications page like GitHub so it's a little different. It looks like they use a referrer URL to mark it serverside (I'm not a developer so that's a guess), which might be an option in this case.

**Comment by Pickysaurus** [comment]
Is this definitely happening for every notification you click on? Or just some of the time? In our testing, an error message would show for a split-second after clicking the notification and the page would reload, but the notification would only fail to mark as read intermittently.

**Comment by billw2012** [comment]
Behaviour isn't totally clear. If I am on the page the notification is on I can click it many times and it never clears. However I just navigated to another page after doing this and the notification was cleared already on that page.

**Comment by billw2012** [comment]
Also I am noting that the count of "new" messages on my mods keeps going up even though I am marking them as read. At some point it becomes a new set of messages, but its not clear under what circumstances.
![image](https://user-images.githubusercontent.com/1453936/112315855-73c7da00-8ca2-11eb-89bc-381f436f24a3.png)

In the case where it says "7 new comments", I actually got 1 new comment, read it, marked it as read. Then it notified me about 2 new comments, then 3, etc, instead of keeping these groups that I responded to separate.
